com.avaloq.tools.ddk.xtext.builder/src/com/avaloq/tools/ddk/xtext/builder/MonitoredClusteringBuilderState.java

```diff
@@ -694,29 +696,31 @@ protected Resource addResource(final Resource resource, final ResourceSet resour
    *          build data, must not be {@code null}
    */
   protected void storeBinaryResource(final Resource resource, final BuildData buildData) {
-    if (isBinaryModelStorageAvailable && resource instanceof StorageAwareResource && ((StorageAwareResource) resource).getResourceStorageFacade() != null
+    if (resource.getResourceSet() == null) {

```

Comment:
we should do this validation after if (isBinaryModelStorageAvailable && resource instanceof StorageAwareResource && ((StorageAwareResource) resource).getResourceStorageFacade() != null
otherwise we could complain about resources we will never save, and that is not needed

Reply:
You're right.